cocotb/binary.py

```diff
@@ -237,12 +237,12 @@ def _adjust_unsigned(self, x):
             else:
                 rv = '0' * (self._n_bits - l) + x
         elif l > self._n_bits:
-            print("WARNING: truncating value to match requested number of bits "
-                  "(%d -> %d)" % (l, self._n_bits))
             if self.big_endian:
                 rv = x[l - self._n_bits:]
             else:
                 rv = x[:l - self._n_bits]
+            warnings.warn("{}-bit value requested, truncating value {} ({} bits) to {}".format(
+                self._n_bits, x, l, rv), category=RuntimeWarning)
```

Comment:
```suggestion
                self._n_bits, x, l, rv), category=RuntimeWarning, stacklevel=n)
```
Where `n > 2` is some integer that makes the warning message show an interesting line to the user.

Reply:
Done.